app/services/idm/fetch-user-roles-and-groups.service.js

```diff
@@ -18,12 +18,12 @@ const UserModel = require('../../models/user.model.js')
  * later added to a group which also includes that role). It also returns an array of groups that the user is a member
  * of, along with `userFound` to explicitly indicate whether or not the user id exists.
  *
- * @param {Number} userId The user id to get roles and groups for
+ * @param {number} userId - The user id to get roles and groups for
  *
- * @returns {Promise<Object>} result The resulting roles and groups
- * @returns {UserModel} result.user Returns the UserModel representing the user, or `null` if the user is not found
- * @returns {RoleModel[]} result.roles An array of RoleModel objects representing the roles the user has
- * @returns {GroupModel[]} result.groups An array of GroupModel objects representing the groups the user is a member of
+ * @returns {Promise<object>} result - A promise that resolves to an object containing:
```

Comment:
The linter enforce single returns.

This is the proposed soultion that is not a typedef.